segmentation_labeling_app/transforms/transformations.py

```diff
@@ -0,0 +1,212 @@
+                           box_size: Tuple[int, int],
+                           video_shape: Tuple[int, int]):
+    """
+    Returns a coordinate pair that has been transformed to fit as closely to
+    original position as possible but still with box contained totally in video.
+    This is made to shift the coordinate if the subset box goes out of bound of
+    the video shape.
+    Args:
+        coordinate_pair: the original coordinate location for centering
+        box_size: the size of the box for subset
+        video_shape: the shape of the original video
+
+    Returns:
+
+    """
+    if box_size[0] < video_shape[0] and box_size[1] < video_shape[1]:
+        # if the box size is less than frame size it can only be out of bounds in
+        # direction left and right and one direction up and down
+        transformed_x = coordinate_pair[0]
+        transformed_y = coordinate_pair[1]
+        # can't be centered left
+        if coordinate_pair[0] < (math.ceil(box_size[0]/2) - 1):
+            transformed_x = math.ceil(box_size[0] / 2) - 1
+
+        # can't be centered up
+        if coordinate_pair[1] < (math.ceil(box_size[1]/2) - 1):
+            transformed_y = math.ceil(box_size[1] / 2) - 1
+
+        # can't be centered right
+        if (coordinate_pair[0] + (math.ceil(box_size[0]/2))) > video_shape[0] - 1:
+            transformed_x = video_shape[0] - math.ceil(box_size[0] / 2) - 1
+
+        # can't be centered down
+        if (coordinate_pair[1] + (math.ceil(box_size[1]/2))) > video_shape[1] - 1:
+            transformed_y = video_shape[1] - math.ceil(box_size[1] / 2) - 1
+
+        return transformed_x, transformed_y
+
+    else:
+        raise ValueError('Box is larger or equal in size in one or more'
+                         'dimensions to original video shape. Provide a '
+                         'box size smaller than original video.')
+
+
+def get_centered_coordinate_box(coordinate_pair: Tuple[int, int],
+                                box_size: Tuple[int, int],
+                                video_array: np.ndarray):
+    """
+    Function to get video subset centered around coordinate pair with
+    size specified by box size tuple. Function takes a subset of each frame
+    and stacks together to get final video.
+    Args:
+        coordinate_pair: the coordinate on which to center the subset
+        box_size: the size of the subset box
+        video_array: the video to take the subset from
+
+    Returns:
+
+    """
+    transformed_coordinates = get_transformed_center(coordinate_pair, box_size,
+                                                     video_array.shape)
+    left_column = transformed_coordinates[0] - math.ceil(box_size[0] / 2)
+    right_column = transformed_coordinates[0] + math.ceil(box_size[0] / 2)
+    up_row = transformed_coordinates[1] - math.ceil(box_size[1] / 2)
+    down_row = transformed_coordinates[1] + math.ceil(box_size[1] / 2)
+    transformed_stack = []
```

Comment:
You're reallocating the memory for this large array 2x -- in this array and then converting it to a numpy array with `stack`. I suggest either allocate a numpy array from the beginning or operate directly on the original array.

Reply:
Good call, I allocated up front and then altered memory as the function loops.